### ISSUE TITLE
Add compression algorithm benchmarks

### ISSUE THREAD

**Issue by jsumners-nr** [RafaelGSS/nodejs-bench-operations#271]
I think it would be helpful to include benchmarks for the common compression algorithms that `zlib` supports. In particular, I would like to see the comparison between `deflate` and `gzip`. Some sort of "large" JSON representation should be used for the test. https://github.com/matthewreagan/WebstersEnglishDictionary/blob/master/dictionary.json may be _too_ large, but something at least a couple of megabytes.

**Comment by RafaelGSS** [comment]
SGTM

**Comment by jsumners-nr** [comment]
I'm not sure this is going to work. I added a `compression.mjs` script like so:

```js
import { fileURLToPath } from 'node:url'
import fs from 'node:fs'
import zlib from 'node:zlib'
import { createBenchmarkSuite } from '../common.mjs'

const fixturePath = fileURLToPath(
  new URL('./fixtures/words_dictionary.json.gz', import.meta.url)
)
let fixtureData = fs.readFileSync(fixturePath)
fixtureData = zlib.gunzipSync(fixtureData)

const suite = createBenchmarkSuite('Compression algorithms')

suite
  .add('Deflate', function () {
    zlib.deflateSync(fixtureData)
  })
  .add('Gzip', function () {
    zlib.gzipSync(fixtureData)
  })

await suite.runAndPrintResults()
```

With the fixture data coming from https://github.com/dwyl/english-words/blob/master/words_dictionary.json (I `gzip -9` that file to keep the repo size down).

The result is always:

```
❯ node --allow-natives-syntax bench/compression.mjs
## Compression algorithms

|name|ops/sec|samples|
|-|-|-|
|Deflate|9|11|
|Gzip|9|11|
```

I don't think that's giving a useful result. But I could be setting up the benchmark incorrectly.

**Comment by RafaelGSS** [comment]
Right,  for this kind of operation, I believe a time-based benchmark would work better. Example: https://github.com/RafaelGSS/nodejs-bench-operations/blob/main/bench/array-append.mjs

**Comment by jsumners-nr** [comment]
Thanks. I'll look into that.

**Comment by RafaelGSS** [comment]
We can now use it through the default benchmark suite, just using time-mode benchmark https://github.com/RafaelGSS/bench-node/pull/71

**Comment by jsumners-nr** [comment]
Awesome. One day I'll get back to it 😂

**Comment by RafaelGSS** [comment]
https://github.com/RafaelGSS/nodejs-bench-operations/pull/288/files
